src/sas/qtgui/Plotting/QRangeSlider.py

```diff
@@ -0,0 +1,236 @@
+"""
+Double slider interactor for setting the Q range for a fit or function
+"""
+import numpy as np
+
+from sas.qtgui.Plotting.PlotterData import Data1D
+from sas.qtgui.Plotting.Slicers.BaseInteractor import BaseInteractor
+
+
+class QRangeSlider(BaseInteractor):
+    """
+    Draw a pair of draggable vertical lines. Each line can be linked to a GUI input.
+    The GUI input should update the lines and vice-versa.
+    """
+    def __init__(self, base, axes, color='black', zorder=5, data=None):
+        """
+        """
+        BaseInteractor.__init__(self, base, axes, color=color)
+        assert isinstance(data, Data1D)
+        self.base = base
+        self.markers = []
+        self.axes = axes
+        self.data = data
+        self.connect = self.base.connect
+        self.x_min = np.fabs(min(self.data.x))
+        self.y_marker_min = self.data.y[np.where(self.data.x == self.x_min)[0][0]]
+        self.x_max = np.fabs(max(self.data.x))
+        self.y_marker_max = self.data.y[np.where(self.data.x == self.x_max)[0][-1]]
+        self.updateOnMove = data.slider_update_on_move
+        self.line_min = LineInteractor(self, axes, zorder=zorder, x=self.x_min, y=self.y_marker_min,
+                                       input=self.data.slider_low_q_input, setter=self.data.slider_low_q_setter,
```

Comment:
You are already passing the `Plotter1D` instance, so the slider related parameters can be stored on a Plotter rather than in Data1D, as mentioned above.